app/controllers/CheckDetailsController.scala

```diff
@@ -173,7 +218,7 @@ class CheckDetailsController @Inject()(
                                           mode: Mode,
                                           sessionWithNINO: Session,
                                           origin: Option[String])(implicit  headerCarrier: HeaderCarrier): Result = {
-    auditService.findYourNinoPDVMatched(pdvData, origin, idData)
+    auditService.findYourNinoPDVMatched(pdvData, origin, Some(idData))
 
     individualDetailsService.createIndividualDetailsData(sessionWithNINO.data.getOrElse("sessionId", EmptyString), idData)
```

Comment:
This line of code returns a future of a string yet we do not wait for the future to complete? What happens if the code tries to update the PDV record (line 226) before this future completes? 

Line 226 then returns a future of a boolean. We also do not wait for this future to complete before proceeding. 

May need to refactor here to ensure these futures complete before proceeding? 

I can take a look on Thursday.